VTUL/dlp-access
# LIBTD-2333: Allow Site Admin to update 4 top-level site configuration…

… fields and record the modification event in the History table

**JIRA Ticket**: (https://webapps.es.vt.edu/jira/browse/LIBTD-2333) (:star:)

Preview is deployed at: https://libtd-2333.djf1n0m63xbku.amplifyapp.com/

# What does this Pull Request do? (:star:)
This PR sets up the Site and History types along with their authorization rules in GraphQL schema to only allow a siteAdmin user to edit site configuration fields as well as recording the modification event into DynamoDB table. This PR only supports the editing capability of 4 top level fields including "analyticsID", "siteColor", "siteName" and "siteTitle".

# What's the changes? (:star:)

* api/collectionarchives/schema.graphql: Add fields into Site type and History type and set up authorization rules
* package.json: add "deep-object-diff" library to extract only the modified fields in the editing form
* src/graphql: changed schema and queries
* App.js: reflect the modified site configuration fields in the application interface
* SiteForm.js: add fields into editing form, extract the modification event info and save it into History table

# How should this be tested?

* amplify push
* Go to the application, sign in/up into siteAdmin page as a properly authenticated user
* Try to edit the 4 site config fields and check if the configuration changes are reflected on the application interface
* Go to DynamoDB table to see if the corresponding modification event has been recorded properly

# Interested parties
Tag (@yinlinchen) interested parties

(:star:) Required fields
